app/models/user.rb

```diff
@@ -102,7 +102,7 @@ def self.from_omniauth(auth)
 
   def self.new_with_session(params, session)
     super.tap do |user|
-      if (data = session['devise.facebook_data'])
+      if (data = session['devise.facebook_data'] || session['devise.google_data'])
```

Comment:
and with each new provider this condition should grow?

Reply:
Yes, it looks not good, but I can't find  a proper solution to make it looks better. I'll think about it..